packages/amplication-container-builder/src/cloud-build/CloudBuildProvider.ts

```diff
@@ -17,8 +26,48 @@ export class CloudBuildProvider implements IProvider {
       projectId: this.projectId,
       build: createConfig(repository, tag, url, buildArgs),
     });
-    // Wait for build to finish
-    const [finishedCloudBuildBuild] = await cloudBuildBuild.promise();
-    return { images: finishedCloudBuildBuild.images as string[] };
+
+    return {
+      status: EnumBuildStatus.Running,
+      statusQuery: { filter: `tags="${tag}"` },
+    };
+  }
+
+  async getStatus(statusQuery: any): Promise<BuildResult> {
+    const data: StatusQuery = statusQuery;
+    const [buildList] = await this.cloudBuild.listBuilds({
+      projectId: this.projectId,
+      filter: data.filter,
+      pageSize: 1,
+    });
+
+    const [build] = buildList;
+    if (!build) {
+      throw new InvalidBuildProviderState(
+        statusQuery,
+        "Can't find the specified build in Cloud Build service"
+      );
+    }
+    switch (build.status) {
+      case BUILD_STATUS_WORKING:
+      case BUILD_STATUS_QUEUED:
+        return {
+          status: EnumBuildStatus.Running,
+          statusQuery: { filter: `build_id="${build.id}"` },
```

Comment:
I prefer to return the query with every result to keep all the responsibility within the provider. 
Otherwise, the build service needs to check with the return result if the query exist or not, or have a separate code path for CreateBuild and getStatus